src/test/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtensionTest.java

```diff
@@ -105,10 +110,12 @@
             + "          c = moshi.adapter(Integer.class).fromJson(reader);\n"
             + "        } else if (\"d\".equals(_name)) {\n"
             + "          d = moshi.adapter(String.class).fromJson(reader);\n"
+            + "        } else if (\"e\".equals(_name)) {\n"
+            + "          e = moshi.adapter(Map.class).fromJson(reader);\n"
```

Comment:
This isn't correct as there's no type information to deserialize the values of the map. You'll get back a `Map<String, Object>` not a `Map<String, Number>` which will ultimately fail at runtime. The latest snapshot of Moshi exposes a `Types` type which allows you to build parameterized types. For example, here you'd want `Types. newParameterizedType(Map.class, String.class, Number.class)`.